Develop

braceal opened the pull request.

braceal commented:
@lee212 I think these are the latest documentation changes. If they look good to you we can merge to main and then set readthedocs to track the main branch. Once that is done, I think we can safely delete the feature/docs branch.

This is in preparation to start merging in the streaming capability.

braceal commented:
@lee212 I merged the issue templates from main into develop so now develop is caught up with the main branch. I also added the installation documentation that Li Tan added to the feature/async branch.

lee212 commented on README.md:
````diff
@@ -19,19 +30,76 @@ pip install --upgrade pip setuptools wheel
+
+```
+conda config --prepend channels https://public.dhe.ibm.com/ibmdl/export/pub/software/server/ibm-ai/conda/
+conda install powerai-rapids
````
Isn't this for IBM PowerPC? other platforms may use `conda install rapids` to get cuML.

braceal replied:
Yes I believe so, but I have not tested it on other platforms. Perhaps we can add documentation for other platforms as we increase our support of those platforms. It might be a good idea to have an installation page in the docs for each platform. Or if not a full page, at least a section within the main installation page.

lee212 commented on README.md:
````diff
@@ -19,19 +30,76 @@ pip install --upgrade pip setuptools wheel
+3. Install packages `scikit-learn` and `h5py` version 2.10.0:
+
+```
+conda install scikit-learn h5py=2.10.0
````
does h5py have to be pinned to 2.10?

braceal replied:
It may have to be to use the virtual H5 API. I have to investigate it more thoroughly but I think for now we should pin to 2.10.0 since that is the version we have tested up to this point and I think I recall having issues with the virtual h5py API with later versions.

There are work arounds to avoid the virtual h5py API, it's something we can update in the future.

lee212 reviewed (approved):
@braceal thanks, Looks good to me for merging this PR, I made minor comments.

braceal commented:
I updated the readthedocs page to track the main branch and updated the readthedocs settings to use the configuration in `docs/source/conf.py`. The build passed on readthedocs.
